PersistentStorage/Sources/Data/Repository/PersistentStorage.swift

```diff
@@ -42,14 +42,19 @@ public final class PersistentStorage {
             forKey: key
         )
         // This operation is always successful
-        return Just(true).eraseToAnyPublisher()
+        return Just(())
+            .setFailureType(to: PersistentStorageError.self)
+            .eraseToAnyPublisher()
     }
 
-    /// TODO
+    /// Method for removing a value with assigned key from the user defaults storage.
+    ///
+    /// - Parameters:
+    ///     - key: Key that will be assigned to value
     @discardableResult
     public func remove(
         valueKey: String
-    ) -> AnyPublisher<Bool, Never> {
+    ) -> AnyPublisher<Void, PersistentStorageError> {
```

Comment:
It doesn't make sense to return error here, if function actually can't fail. For me it's better to remain failure type as `Never` and leave `setFailureType` outside when needed. Maybe I don't see whole context, so tell me more if I am wrong. 😄

Reply:
When you call this function in `Never` failure flow, you have to somehow handle error from this function, even if it can't return error anyway.

Reply:
Second reason for leaving failure type as `Never` is, if you use this remove function in another function with different error type, you have to explicitly map this non sense error to this new one.